planetiler-custommap/src/test/java/com/onthegomap/planetiler/custommap/ConfiguredSingleFeatureTest.java

```diff
@@ -0,0 +1,127 @@
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.yaml.snakeyaml.Yaml;
+
+public class ConfiguredSingleFeatureTest {
+
+  private static FeatureCollector polygonFeatureCollector() {
+    PlanetilerConfig config = PlanetilerConfig.defaults();
+    FeatureCollector.Factory factory = new FeatureCollector.Factory(config, Stats.inMemory());
+    return factory.get(SimpleFeature.create(TestUtils.newPolygon(0, 0, 0.1, 0, 0.1, 0.1, 0, 0), new HashMap<>()));
+  }
+
+  private static FeatureCollector linestringFeatureCollector() {
+    PlanetilerConfig config = PlanetilerConfig.defaults();
+    FeatureCollector.Factory factory = new FeatureCollector.Factory(config, Stats.inMemory());
+    return factory.get(SimpleFeature.create(TestUtils.newLineString(0, 0, 0.1, 0, 0.1, 0.1, 0, 0), new HashMap<>()));
+  }
+
+  private static Profile configureProfile(String filename) throws Exception {
+    Path staticAttributeConfig = TestConfigurableUtils.pathToResource(filename);
+    Yaml yml = new Yaml();
+    SchemaConfig schema = yml.loadAs(new FileInputStream(staticAttributeConfig.toFile()), SchemaConfig.class);
+
+    ConfiguredProfile profile = new ConfiguredProfile(schema);
+    return profile;
+  }
+
+  private static void testFeature(String profileConfig, SourceFeature sf, Supplier<FeatureCollector> fcFactory,
+    Consumer<Feature> test)
+    throws Exception {
+
+    Profile profile = configureProfile(profileConfig);
+    FeatureCollector fc = fcFactory.get();
+
+    profile.processFeature(sf, fc);
+
+    AtomicInteger length = new AtomicInteger(0);
+
+    fc.forEach(f -> {
+      test.accept(f);
+      length.incrementAndGet();
+    });
+
+    assertEquals(1, length.get());
+  }
+
+  private static void testPolygon(String profileConfig, Map<String, Object> tags, Consumer<Feature> test)
+    throws Exception {
+    SourceFeature sf = new TestAreaSourceFeature(tags, "osm", "testLayer");
+    testFeature(profileConfig, sf, ConfiguredSingleFeatureTest::polygonFeatureCollector, test);
+  }
+
+  private static void testLinestring(String profileConfig, Map<String, Object> tags, Consumer<Feature> test)
+    throws Exception {
+    SourceFeature sf = new TestLinestringSourceFeature(tags, "osm", "testLayer");
+    testFeature(profileConfig, sf, ConfiguredSingleFeatureTest::linestringFeatureCollector, test);
+  }
+
+  private static Map<String, Object> waterTags = new HashMap<>();
+  private static Map<String, Object> motorwayTags = new HashMap<>();
+
+  @BeforeAll
+  private static void setup() {
+    waterTags.put("natural", "water");
+    waterTags.put("water", "pond");
+    waterTags.put("name", "Little Pond");
+
+    motorwayTags.put("highway", "motorway");
+    motorwayTags.put("layer", "1");
+    motorwayTags.put("bridge", "yes");
+  }
```

Comment:
this could be simplified to:

```java
  private static final Map<String, Object> waterTags = Map.of(
    "natural", "water",
    "water", "pond",
    "name", "Little Pond"
  );
```

Reply:
Fixed in c5162ef
Yet another post-Java 8 functionality I wasn't aware of!